src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertiesShouldNotBeWriteOnly.cs

```diff
@@ -36,14 +35,45 @@ public abstract class PropertiesShouldNotBeWriteOnlyAnalyzer : DiagnosticAnalyze
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182165.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(AddGetterRule, MakeMoreAccessibleRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterSymbolAction(symbolContext => AnalyzeSymbol(symbolContext), SymbolKind.Property);
+        }
+
+        /// <summary>
+        /// Implementation for CA1044: Properties should not be write only
+        /// </summary>
+        /// <param name="context"></param>
+        private static void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+
+            var property = context.Symbol as IPropertySymbol;
+            // If no property found
+            if (property == null)
```

Comment:
You don't need this check. Since you register for property symbols, you will only be passed back property symbols.